ksql-engine/src/main/java/io/confluent/ksql/topic/TopicDeleteInjector.java

```diff
@@ -99,22 +103,21 @@ public TopicDeleteInjector(
             () -> topicClient.deleteTopics(ImmutableList.of(source.getKafkaTopicName())),
             ExecutorUtil.RetryBehaviour.ALWAYS);
       } catch (Exception e) {
-        throw new KsqlException("Could not delete the corresponding kafka topic: "
-            + source.getKafkaTopicName(), e);
+        throw new RuntimeException("Could not delete the corresponding kafka topic: "
+                + sourceName, e);
       }
 
-      if (source.getValueSerdeFactory().getFormat() == Format.AVRO) {
-        try {
+      try {
+        if (source.getValueSerdeFactory().getFormat() == Format.AVRO) {
           SchemaRegistryUtil.deleteSubjectWithRetries(
-              schemaRegistryClient,
-              source.getKafkaTopicName() + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
-        } catch (final Exception e) {
-          throw new KsqlException("Could not clean up the schema registry for topic: "
-              + source.getKafkaTopicName(), e);
+                  schemaRegistryClient,
+                  source.getKafkaTopicName() + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
         }
+      } catch (final Exception e) {
+        checkSchemaError(e, source.getKafkaTopicName());
       }
     } else if (dropStatement.getIfExists()) {
-      throw new KsqlException("Could not find source to delete topic for: " + statement);
```

Comment:
This is a client error - they've asked us to check a condition before executing which evaluates to false. So KsqlException would be appropriate in this case.